internal/cache/profiles.go

```diff
@@ -190,18 +190,28 @@ func (p *profileCache) DeviceResource(profileName string, resourceName string) (
 
 // CommandExists returns a bool indicating whether the specified command exists for the
 // specified (by name) device. If the specified device doesn't exist, an error is returned.
-func (p *profileCache) CommandExists(profileName string, cmd string) (bool, error) {
+func (p *profileCache) CommandExists(profileName string, cmd string, method string) (bool, error) {
 	p.mutex.Lock()
 	defer p.mutex.Unlock()
 
-	commands, ok := p.ccMap[profileName]
-	if !ok {
+	coreCommands, profileExist := p.ccMap[profileName]
+	if !profileExist {
 		err := fmt.Errorf("specified profile: %s not found", profileName)
 		return false, err
 	}
+	// Check whether cmd exists in either coreCommands or deviceCommands.
+	if _, ccExist := coreCommands[cmd]; !ccExist {
```

Comment:
So this allows for the device commands to be eliminated when we have Core Commands?

Reply:
I'm not sure if you're referring to be eliminated from device profile .. ?
In this case, yes, device commands and core commands are both optional in device profile and by definition core commands are **"commands which map one to one to resource (commands)."** so if core commands map to the correct resource, device commands can be eliminated.

I'm trying to do here is to double check command doesn't exist in core command map or device command map (or you can say doesn't exist in device profile)